{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "import customfunc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "class primer_list ():\n",
    "    \n",
    "    \n",
    "    def __init__(self,diretorio):\n",
    "        import glob\n",
    "        import os\n",
    "        from Bio import SeqIO\n",
    "        self.diretorio = diretorio\n",
    "        abi = glob.glob(diretorio + \"/*.ab1\")\n",
    "        self.abi = abi\n",
    "        if len(abi) == 0:\n",
    "            print('Lista de abi vazia. Escolha outro diretorio')\n",
    "            \n",
    "    def to_fasta(self,NomePasta):\n",
    "        import glob\n",
    "        import os\n",
    "        from Bio import SeqIO\n",
    "        os.system(f'mkdir {self.diretorio}/{NomePasta}')\n",
    "        \n",
    "        for arquivos_abi in self.abi:\n",
    "            f = SeqIO.convert(f'{arquivos_abi}', 'abi', f'{arquivos_abi}.fasta','fasta')\n",
    "            os.system(f'mv {arquivos_abi}.fasta {NomePasta}')\n",
    "        fastas = glob.glob(f\"{self.diretorio}/{NomePasta}\" + '/*.fasta')\n",
    "        \n",
    "        for arquivos in fastas:\n",
    "            for nomes in SeqIO.parse(f'{arquivos}','fasta'):\n",
    "                nome = arquivos.split('/')[-1].split('.')[0] + '.' + arquivos.split('/')[-1].split('.')[1]\n",
    "                novo = open(f'{nome}.fasta','a')\n",
    "                ident = nome.split('.')[0]\n",
    "                novo.write(f'>{ident}\\n')\n",
    "                novo.write(f'{str(nomes.seq)}')\n",
    "                novo.close()\n",
    "                os.system(f'rm {arquivos}')\n",
    "                os.system(f'mv {nome}.fasta {self.diretorio}/{NomePasta}')\n",
    "                \n",
    "        fastas = glob.glob(f\"{self.diretorio}/{NomePasta}\" + '/*.fasta')\n",
    "        \n",
    "        if len(fastas) == 0:\n",
    "            print('pasta com fastas vazia')\n",
    "        else:\n",
    "            self.fastas = fastas\n",
    "            \n",
    "        forward = [f for f in self.fastas if \"F\" in f.split('/')[-1].split('.')[0]]\n",
    "        self.forward = forward\n",
    "        forward.sort()\n",
    "        \n",
    "        reverse = [r for r in self.fastas if \"R\" in r.split('/')[-1].split('.')[0]]\n",
    "        self.reverse = reverse\n",
    "        reverse.sort()\n",
    "        \n",
    "        if len(forward) != len(forward):\n",
    "            print('As listas forward e reverse tem tamanhos diferentes')            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "class blast_sequences ():\n",
    "    \n",
    "    def __init__(self,f,r,diretorio):\n",
    "        import os\n",
    "        self.r = r\n",
    "        self.f = f\n",
    "        self.diretorio = diretorio\n",
    "        \n",
    "        \n",
    "    def blast(self,modo,genoma,*lista):\n",
    "        from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "        import os\n",
    "        self.lista = lista\n",
    "        self.modo = str(modo).upper()\n",
    "        self.genoma = genoma\n",
    "        os.system(f'mkdir {self.diretorio}')\n",
    "        \n",
    "        if modo == '_F':\n",
    "            for primers in self.f:\n",
    "                for identificadores in self.lista:\n",
    "                    if primers.split('/')[-1].split('.')[0].endswith(identificadores):\n",
    "                        c = NcbiblastnCommandline(query = f'{primers}', subject = genoma ,out = f\"{primers}_{genoma}.csv\", outfmt = 6)\n",
    "                        c()\n",
    "                        os.system(f'mv {primers}_{genoma}.csv {self.diretorio}')\n",
    "                   \n",
    "                    \n",
    "        elif modo == '_R':\n",
    "            for primers in self.r:\n",
    "                for identificadores in self.lista:\n",
    "                    if primers.split('/')[-1].split('.')[0].endswith(identificadores):\n",
    "                        c = NcbiblastnCommandline(query = f'{primers}', subject = genoma ,out = f\"{primers}_{genoma}.csv\", outfmt = 6)\n",
    "                        c()\n",
    "                        os.system(f'mv {primers}_{genoma}.csv {self.diretorio}')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "primers = primer_list('/home/tiagoboreli/Documentos/trimopython')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/tiagoboreli/Documentos/trimopython/Gabriel_F_55.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_28.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_49.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1153_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_57.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_51.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_30.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_26.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_84.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1183_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1155_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1149_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_56.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_62.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1154_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1155_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_78.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_75.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_34.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_31.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1181_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_76.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_11.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1181_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1121_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1130_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_55.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1122_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_36.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1159_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_46.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_76.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_09.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1127_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1146_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_70.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_73.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_46.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_50.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_06.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_11.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_09.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_18.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_60.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1154_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_50.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_77.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_64.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_10.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_27.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1150_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_26.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_29.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1131_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_67.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_61.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_62.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_66.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_02.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_23.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_52.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_79.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_89.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_37.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_58.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1184_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_63.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1150_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_30.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_78.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_60.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_42.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_18.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_31.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_02.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_73.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1123_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_49.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_59.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_34.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_27.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_28.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_14.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_12.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_51.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_22.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1190_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1183_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_61.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_10.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1153_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_69.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1149_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_64.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_68.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_84.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_63.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_65.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1131_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_22.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_12.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_06.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1189_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_79.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_37.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_23.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_59.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_70.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1132_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_75.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1159_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_74.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_42.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_71.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_01.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_01.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1123_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1190_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1189_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_72.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_56.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_89.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_57.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_65.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_80.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_48.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1122_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_80.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_36.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1127_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1132_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1146_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_58.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_48.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1121_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1130_F.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_52.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_95.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_66.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_67.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_71.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_74.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_72.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_68.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_1184_R.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_29.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_69.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_14.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_F_95.ab1',\n",
       " '/home/tiagoboreli/Documentos/trimopython/Gabriel_R_77.ab1']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primers.abi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "primers.to_fasta('fastas')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1121_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1122_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1123_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1127_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1130_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1131_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1132_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1146_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1149_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1150_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1153_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1154_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1155_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1159_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1181_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1183_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1184_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1189_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1190_F.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_01.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_02.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_06.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_09.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_10.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_11.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_12.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_14.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_18.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_22.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_23.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_26.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_27.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_28.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_29.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_30.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_31.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_34.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_36.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_37.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_42.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_46.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_48.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_49.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_50.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_51.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_52.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_55.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_56.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_57.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_58.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_59.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_60.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_61.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_62.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_63.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_64.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_65.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_66.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_67.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_68.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_69.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_70.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_71.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_72.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_73.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_74.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_75.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_76.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_77.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_78.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_79.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_80.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_84.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_89.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_F_95.ab1.fasta']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primers.forward"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1121_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1122_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1123_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1127_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1130_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1131_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1132_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1146_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1149_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1150_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1153_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1154_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1155_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1159_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1181_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1183_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1184_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1189_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_1190_R.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_01.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_02.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_06.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_09.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_10.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_11.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_12.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_14.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_18.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_22.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_23.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_26.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_27.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_28.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_29.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_30.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_31.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_34.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_36.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_37.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_42.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_46.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_48.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_49.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_50.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_51.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_52.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_55.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_56.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_57.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_58.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_59.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_60.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_61.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_62.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_63.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_64.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_65.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_66.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_67.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_68.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_69.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_70.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_71.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_72.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_73.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_74.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_75.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_76.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_77.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_78.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_79.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_80.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_84.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_89.ab1.fasta',\n",
       " '/home/tiagoboreli/Documentos/trimopython/fastas/Gabriel_R_95.ab1.fasta']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primers.reverse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(primers.reverse) == len(primers.forward)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast = blast_sequences(f = primers.forward, r = primers.reverse, diretorio = '/home/tiagoboreli/Documentos/trimopython/blasts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast.blast('_F','genome36.fasta','1121_F','1122_F','1123_F', '1127_F', '1130_F', '1131_F', '1132_F', '1146_F', '1149_F', '1150_F', '1153_F', '1154_F', '1155_F')\n",
    "blast.blast('_R','genome36.fasta','1121_R','1122_R','1123_R', '1127_R', '1130_R', '1131_R', '1132_R', '1146_R', '1149_R', '1150_R', '1153_R', '1154_R', '1155_R')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast.blast('_F', 'genome40.fasta','1159_F','1181_F','1183_F','1184_F','1189_F','1190_F','_22','_23')\n",
    "blast.blast('_R', 'genome40.fasta','1159_R','1181_R','1183_R','1184_R','1189_R','1190_R','_22','_23')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast.blast('_F','genome3.fasta','_01','_02','_27','_28','_29','_37','_42','_46','_48','_49','_51','_55','_56','_57','_58','_59','_60','_61','_63','_67','_68','_72','_73','_74','_75','_76','_77','_78','_79','_80','_84','_89','_95')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast.blast('_R','genome3.fasta','_01','_02','_27','_28','_29','_37','_42','_46','_48','_49','_51','_55','_56','_57','_58','_59','_60','_61','_63','_67','_68','_72','_73','_74','_75','_76','_77','_78','_79','_80','_84','_89','_95')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "arquivos =  glob.glob(\"/home/tiagoboreli/Documentos/trimopython/blasts\" + '/*.csv')\n",
    "lista_df = []\n",
    "for arqs in arquivos:\n",
    "    df = pd.read_csv(arqs, index_col=None, header=None, sep = '\\t',nrows = 1 ,names = ['query acc.ver', 'subject acc.ver','identity', 'alignment length', 'mismatches',' gap opens', 'q. start', 'q. end', 's. start', 's. end', 'evalue', 'bit score'])\n",
    "    df['arquivo'] = arqs\n",
    "    lista_df.append(df)\n",
    "genomes = pd.concat(lista_df,axis=0, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query acc.ver</th>\n",
       "      <th>subject acc.ver</th>\n",
       "      <th>identity</th>\n",
       "      <th>alignment length</th>\n",
       "      <th>mismatches</th>\n",
       "      <th>gap opens</th>\n",
       "      <th>q. start</th>\n",
       "      <th>q. end</th>\n",
       "      <th>s. start</th>\n",
       "      <th>s. end</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bit score</th>\n",
       "      <th>arquivo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gabriel_1150_F</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>95.072</td>\n",
       "      <td>1116</td>\n",
       "      <td>20</td>\n",
       "      <td>34</td>\n",
       "      <td>39</td>\n",
       "      <td>1151</td>\n",
       "      <td>1</td>\n",
       "      <td>1084</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1724</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gabriel_F_51</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>98.082</td>\n",
       "      <td>730</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>51</td>\n",
       "      <td>776</td>\n",
       "      <td>17064</td>\n",
       "      <td>17787</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1262</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gabriel_R_58</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>98.276</td>\n",
       "      <td>406</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>33</td>\n",
       "      <td>438</td>\n",
       "      <td>19645</td>\n",
       "      <td>20047</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>708</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gabriel_1122_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>97.528</td>\n",
       "      <td>890</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>33</td>\n",
       "      <td>909</td>\n",
       "      <td>3556</td>\n",
       "      <td>2669</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1507</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gabriel_1146_F</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>96.992</td>\n",
       "      <td>1064</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>32</td>\n",
       "      <td>1074</td>\n",
       "      <td>1</td>\n",
       "      <td>1063</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1768</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Gabriel_F_58</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.073</td>\n",
       "      <td>1044</td>\n",
       "      <td>6</td>\n",
       "      <td>34</td>\n",
       "      <td>9</td>\n",
       "      <td>1051</td>\n",
       "      <td>23247</td>\n",
       "      <td>22238</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1668</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Gabriel_1122_F</td>\n",
       "      <td>NODE_102_length_736_cov_307.748823</td>\n",
       "      <td>98.641</td>\n",
       "      <td>368</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>430</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>652</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Gabriel_F_01</td>\n",
       "      <td>NODE_4_length_193532_cov_12.310185</td>\n",
       "      <td>94.130</td>\n",
       "      <td>954</td>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>31</td>\n",
       "      <td>984</td>\n",
       "      <td>138839</td>\n",
       "      <td>139766</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1428</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Gabriel_1146_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>96.958</td>\n",
       "      <td>1052</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>32</td>\n",
       "      <td>1080</td>\n",
       "      <td>2387</td>\n",
       "      <td>1354</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1746</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Gabriel_1183_R</td>\n",
       "      <td>NODE_51_length_2906_cov_7.795155</td>\n",
       "      <td>96.920</td>\n",
       "      <td>1104</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>27</td>\n",
       "      <td>1130</td>\n",
       "      <td>1842</td>\n",
       "      <td>765</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1827</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Gabriel_R_61</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.952</td>\n",
       "      <td>853</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>35</td>\n",
       "      <td>887</td>\n",
       "      <td>21409</td>\n",
       "      <td>20582</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1408</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Gabriel_F_79</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.734</td>\n",
       "      <td>1041</td>\n",
       "      <td>4</td>\n",
       "      <td>29</td>\n",
       "      <td>50</td>\n",
       "      <td>1090</td>\n",
       "      <td>18667</td>\n",
       "      <td>19677</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1707</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Gabriel_R_76</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.917</td>\n",
       "      <td>746</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>52</td>\n",
       "      <td>797</td>\n",
       "      <td>23757</td>\n",
       "      <td>23033</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1230</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gabriel_R_67</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.530</td>\n",
       "      <td>778</td>\n",
       "      <td>3</td>\n",
       "      <td>23</td>\n",
       "      <td>54</td>\n",
       "      <td>831</td>\n",
       "      <td>23757</td>\n",
       "      <td>23004</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1266</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Gabriel_1183_F</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903</td>\n",
       "      <td>94.670</td>\n",
       "      <td>863</td>\n",
       "      <td>7</td>\n",
       "      <td>37</td>\n",
       "      <td>25</td>\n",
       "      <td>848</td>\n",
       "      <td>1083</td>\n",
       "      <td>221</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1303</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Gabriel_F_48</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.285</td>\n",
       "      <td>996</td>\n",
       "      <td>9</td>\n",
       "      <td>28</td>\n",
       "      <td>26</td>\n",
       "      <td>1020</td>\n",
       "      <td>19357</td>\n",
       "      <td>20325</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1609</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Gabriel_1159_F</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>95.556</td>\n",
       "      <td>1125</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>32</td>\n",
       "      <td>1154</td>\n",
       "      <td>118155</td>\n",
       "      <td>119257</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1779</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Gabriel_1159_R</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>96.597</td>\n",
       "      <td>1146</td>\n",
       "      <td>6</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>1174</td>\n",
       "      <td>120091</td>\n",
       "      <td>118979</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1869</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Gabriel_R_51</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.472</td>\n",
       "      <td>791</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>29</td>\n",
       "      <td>818</td>\n",
       "      <td>22450</td>\n",
       "      <td>21676</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1334</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Gabriel_1154_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>99.109</td>\n",
       "      <td>449</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>482</td>\n",
       "      <td>3251</td>\n",
       "      <td>2803</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>806</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Gabriel_R_59</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.540</td>\n",
       "      <td>691</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>29</td>\n",
       "      <td>719</td>\n",
       "      <td>21574</td>\n",
       "      <td>20899</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1168</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Gabriel_1155_R</td>\n",
       "      <td>NODE_34_length_63067_cov_25.930044</td>\n",
       "      <td>97.803</td>\n",
       "      <td>1138</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>32</td>\n",
       "      <td>1163</td>\n",
       "      <td>61716</td>\n",
       "      <td>60588</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1949</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Gabriel_F_80</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.099</td>\n",
       "      <td>974</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>15</td>\n",
       "      <td>988</td>\n",
       "      <td>19512</td>\n",
       "      <td>20448</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1554</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Gabriel_1189_F</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>97.267</td>\n",
       "      <td>1171</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>48</td>\n",
       "      <td>1210</td>\n",
       "      <td>118155</td>\n",
       "      <td>119306</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1960</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Gabriel_F_75</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>94.707</td>\n",
       "      <td>1058</td>\n",
       "      <td>7</td>\n",
       "      <td>48</td>\n",
       "      <td>4</td>\n",
       "      <td>1060</td>\n",
       "      <td>23137</td>\n",
       "      <td>22128</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1598</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Gabriel_1149_F</td>\n",
       "      <td>NODE_102_length_736_cov_307.748823</td>\n",
       "      <td>98.641</td>\n",
       "      <td>368</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "      <td>407</td>\n",
       "      <td>368</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>652</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Gabriel_R_23</td>\n",
       "      <td>NODE_51_length_2906_cov_7.795155</td>\n",
       "      <td>95.513</td>\n",
       "      <td>780</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>29</td>\n",
       "      <td>808</td>\n",
       "      <td>1842</td>\n",
       "      <td>1097</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1216</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Gabriel_R_42</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>98.300</td>\n",
       "      <td>647</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>37</td>\n",
       "      <td>683</td>\n",
       "      <td>22699</td>\n",
       "      <td>22064</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1123</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Gabriel_1149_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>97.822</td>\n",
       "      <td>1148</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>34</td>\n",
       "      <td>1178</td>\n",
       "      <td>3556</td>\n",
       "      <td>2428</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1962</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Gabriel_1127_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>95.772</td>\n",
       "      <td>1088</td>\n",
       "      <td>24</td>\n",
       "      <td>22</td>\n",
       "      <td>51</td>\n",
       "      <td>1135</td>\n",
       "      <td>1</td>\n",
       "      <td>1069</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1735</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>Gabriel_1130_F</td>\n",
       "      <td>NODE_80_length_2380_cov_29.944761</td>\n",
       "      <td>95.712</td>\n",
       "      <td>583</td>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>170</td>\n",
       "      <td>731</td>\n",
       "      <td>1</td>\n",
       "      <td>583</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>918</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>Gabriel_F_74</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.976</td>\n",
       "      <td>926</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>936</td>\n",
       "      <td>21484</td>\n",
       "      <td>20585</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1530</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Gabriel_1184_R</td>\n",
       "      <td>NODE_51_length_2906_cov_7.795155</td>\n",
       "      <td>97.649</td>\n",
       "      <td>1106</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "      <td>28</td>\n",
       "      <td>1133</td>\n",
       "      <td>1842</td>\n",
       "      <td>758</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1879</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Gabriel_R_89</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>94.504</td>\n",
       "      <td>1019</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>34</td>\n",
       "      <td>1050</td>\n",
       "      <td>19668</td>\n",
       "      <td>20642</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1530</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Gabriel_1150_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>99.641</td>\n",
       "      <td>836</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>883</td>\n",
       "      <td>3251</td>\n",
       "      <td>2418</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1526</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Gabriel_F_84</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.551</td>\n",
       "      <td>1021</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>22</td>\n",
       "      <td>1042</td>\n",
       "      <td>19272</td>\n",
       "      <td>20274</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1731</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Gabriel_R_48</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.678</td>\n",
       "      <td>873</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>33</td>\n",
       "      <td>905</td>\n",
       "      <td>18976</td>\n",
       "      <td>19819</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1424</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Gabriel_R_84</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>95.172</td>\n",
       "      <td>932</td>\n",
       "      <td>2</td>\n",
       "      <td>39</td>\n",
       "      <td>36</td>\n",
       "      <td>967</td>\n",
       "      <td>23757</td>\n",
       "      <td>22869</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1432</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Gabriel_R_02</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.162</td>\n",
       "      <td>912</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>15</td>\n",
       "      <td>926</td>\n",
       "      <td>19668</td>\n",
       "      <td>20546</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1459</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Gabriel_1181_F</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>97.815</td>\n",
       "      <td>1144</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>27</td>\n",
       "      <td>1165</td>\n",
       "      <td>118155</td>\n",
       "      <td>119287</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1960</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Gabriel_R_80</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>95.169</td>\n",
       "      <td>890</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>37</td>\n",
       "      <td>926</td>\n",
       "      <td>21302</td>\n",
       "      <td>20456</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1365</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Gabriel_F_02</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>86.099</td>\n",
       "      <td>446</td>\n",
       "      <td>13</td>\n",
       "      <td>49</td>\n",
       "      <td>52</td>\n",
       "      <td>448</td>\n",
       "      <td>21299</td>\n",
       "      <td>20854</td>\n",
       "      <td>2.620000e-122</td>\n",
       "      <td>435</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Gabriel_R_46</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>95.961</td>\n",
       "      <td>916</td>\n",
       "      <td>6</td>\n",
       "      <td>31</td>\n",
       "      <td>34</td>\n",
       "      <td>949</td>\n",
       "      <td>23236</td>\n",
       "      <td>22352</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1458</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Gabriel_1184_F</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903</td>\n",
       "      <td>98.983</td>\n",
       "      <td>1082</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "      <td>1121</td>\n",
       "      <td>1083</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1929</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Gabriel_1153_R</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>94.227</td>\n",
       "      <td>1126</td>\n",
       "      <td>23</td>\n",
       "      <td>40</td>\n",
       "      <td>9</td>\n",
       "      <td>1133</td>\n",
       "      <td>2356</td>\n",
       "      <td>1272</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1681</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Gabriel_F_23</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903</td>\n",
       "      <td>96.325</td>\n",
       "      <td>1034</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>1040</td>\n",
       "      <td>1084</td>\n",
       "      <td>80</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1672</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Gabriel_R_77</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>95.490</td>\n",
       "      <td>776</td>\n",
       "      <td>4</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>804</td>\n",
       "      <td>21568</td>\n",
       "      <td>20824</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1210</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Gabriel_R_74</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.360</td>\n",
       "      <td>606</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>34</td>\n",
       "      <td>639</td>\n",
       "      <td>19668</td>\n",
       "      <td>20261</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1020</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Gabriel_1154_F</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>98.355</td>\n",
       "      <td>1094</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>42</td>\n",
       "      <td>1134</td>\n",
       "      <td>1</td>\n",
       "      <td>1083</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1910</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Gabriel_F_42</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>95.150</td>\n",
       "      <td>969</td>\n",
       "      <td>6</td>\n",
       "      <td>37</td>\n",
       "      <td>19</td>\n",
       "      <td>987</td>\n",
       "      <td>20028</td>\n",
       "      <td>20955</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1491</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Gabriel_F_37</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>94.523</td>\n",
       "      <td>986</td>\n",
       "      <td>6</td>\n",
       "      <td>47</td>\n",
       "      <td>5</td>\n",
       "      <td>986</td>\n",
       "      <td>22344</td>\n",
       "      <td>21403</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1478</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Gabriel_F_73</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.967</td>\n",
       "      <td>989</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>1001</td>\n",
       "      <td>17067</td>\n",
       "      <td>18029</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1635</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Gabriel_1189_R</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>96.098</td>\n",
       "      <td>1179</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>33</td>\n",
       "      <td>1211</td>\n",
       "      <td>120091</td>\n",
       "      <td>118947</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1892</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Gabriel_1132_F</td>\n",
       "      <td>NODE_74_length_3963_cov_27.927277</td>\n",
       "      <td>91.128</td>\n",
       "      <td>541</td>\n",
       "      <td>27</td>\n",
       "      <td>21</td>\n",
       "      <td>71</td>\n",
       "      <td>591</td>\n",
       "      <td>1</td>\n",
       "      <td>540</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>713</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Gabriel_F_49</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.436</td>\n",
       "      <td>1014</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>1029</td>\n",
       "      <td>21577</td>\n",
       "      <td>20582</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1711</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>Gabriel_R_22</td>\n",
       "      <td>NODE_6_length_238953_cov_6.622376</td>\n",
       "      <td>95.352</td>\n",
       "      <td>710</td>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>40</td>\n",
       "      <td>749</td>\n",
       "      <td>120047</td>\n",
       "      <td>119357</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1110</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Gabriel_1190_R</td>\n",
       "      <td>NODE_51_length_2906_cov_7.795155</td>\n",
       "      <td>97.693</td>\n",
       "      <td>1127</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>26</td>\n",
       "      <td>1151</td>\n",
       "      <td>1842</td>\n",
       "      <td>735</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1919</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Gabriel_R_60</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>96.456</td>\n",
       "      <td>790</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>823</td>\n",
       "      <td>22061</td>\n",
       "      <td>21299</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1279</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Gabriel_F_28</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>97.428</td>\n",
       "      <td>1011</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>57</td>\n",
       "      <td>1064</td>\n",
       "      <td>17064</td>\n",
       "      <td>18054</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1701</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Gabriel_F_56</td>\n",
       "      <td>NODE_40_length_39401_cov_23.095466</td>\n",
       "      <td>91.939</td>\n",
       "      <td>980</td>\n",
       "      <td>30</td>\n",
       "      <td>47</td>\n",
       "      <td>41</td>\n",
       "      <td>1018</td>\n",
       "      <td>21880</td>\n",
       "      <td>20948</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1327</td>\n",
       "      <td>/home/tiagoboreli/Documentos/trimopython/blast...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>108 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      query acc.ver                     subject acc.ver  identity  \\\n",
       "0    Gabriel_1150_F   NODE_74_length_3963_cov_27.927277    95.072   \n",
       "1      Gabriel_F_51  NODE_40_length_39401_cov_23.095466    98.082   \n",
       "2      Gabriel_R_58  NODE_40_length_39401_cov_23.095466    98.276   \n",
       "3    Gabriel_1122_R   NODE_74_length_3963_cov_27.927277    97.528   \n",
       "4    Gabriel_1146_F   NODE_74_length_3963_cov_27.927277    96.992   \n",
       "5      Gabriel_F_58  NODE_40_length_39401_cov_23.095466    96.073   \n",
       "6    Gabriel_1122_F  NODE_102_length_736_cov_307.748823    98.641   \n",
       "7      Gabriel_F_01  NODE_4_length_193532_cov_12.310185    94.130   \n",
       "8    Gabriel_1146_R   NODE_74_length_3963_cov_27.927277    96.958   \n",
       "9    Gabriel_1183_R    NODE_51_length_2906_cov_7.795155    96.920   \n",
       "10     Gabriel_R_61  NODE_40_length_39401_cov_23.095466    96.952   \n",
       "11     Gabriel_F_79  NODE_40_length_39401_cov_23.095466    96.734   \n",
       "12     Gabriel_R_76  NODE_40_length_39401_cov_23.095466    96.917   \n",
       "13     Gabriel_R_67  NODE_40_length_39401_cov_23.095466    96.530   \n",
       "14   Gabriel_1183_F   NODE_40_length_14359_cov_7.162903    94.670   \n",
       "15     Gabriel_F_48  NODE_40_length_39401_cov_23.095466    96.285   \n",
       "16   Gabriel_1159_F   NODE_6_length_238953_cov_6.622376    95.556   \n",
       "17   Gabriel_1159_R   NODE_6_length_238953_cov_6.622376    96.597   \n",
       "18     Gabriel_R_51  NODE_40_length_39401_cov_23.095466    97.472   \n",
       "19   Gabriel_1154_R   NODE_74_length_3963_cov_27.927277    99.109   \n",
       "20     Gabriel_R_59  NODE_40_length_39401_cov_23.095466    97.540   \n",
       "21   Gabriel_1155_R  NODE_34_length_63067_cov_25.930044    97.803   \n",
       "22     Gabriel_F_80  NODE_40_length_39401_cov_23.095466    96.099   \n",
       "23   Gabriel_1189_F   NODE_6_length_238953_cov_6.622376    97.267   \n",
       "24     Gabriel_F_75  NODE_40_length_39401_cov_23.095466    94.707   \n",
       "25   Gabriel_1149_F  NODE_102_length_736_cov_307.748823    98.641   \n",
       "26     Gabriel_R_23    NODE_51_length_2906_cov_7.795155    95.513   \n",
       "27     Gabriel_R_42  NODE_40_length_39401_cov_23.095466    98.300   \n",
       "28   Gabriel_1149_R   NODE_74_length_3963_cov_27.927277    97.822   \n",
       "29   Gabriel_1127_R   NODE_74_length_3963_cov_27.927277    95.772   \n",
       "..              ...                                 ...       ...   \n",
       "78   Gabriel_1130_F   NODE_80_length_2380_cov_29.944761    95.712   \n",
       "79     Gabriel_F_74  NODE_40_length_39401_cov_23.095466    96.976   \n",
       "80   Gabriel_1184_R    NODE_51_length_2906_cov_7.795155    97.649   \n",
       "81     Gabriel_R_89  NODE_40_length_39401_cov_23.095466    94.504   \n",
       "82   Gabriel_1150_R   NODE_74_length_3963_cov_27.927277    99.641   \n",
       "83     Gabriel_F_84  NODE_40_length_39401_cov_23.095466    97.551   \n",
       "84     Gabriel_R_48  NODE_40_length_39401_cov_23.095466    96.678   \n",
       "85     Gabriel_R_84  NODE_40_length_39401_cov_23.095466    95.172   \n",
       "86     Gabriel_R_02  NODE_40_length_39401_cov_23.095466    96.162   \n",
       "87   Gabriel_1181_F   NODE_6_length_238953_cov_6.622376    97.815   \n",
       "88     Gabriel_R_80  NODE_40_length_39401_cov_23.095466    95.169   \n",
       "89     Gabriel_F_02  NODE_40_length_39401_cov_23.095466    86.099   \n",
       "90     Gabriel_R_46  NODE_40_length_39401_cov_23.095466    95.961   \n",
       "91   Gabriel_1184_F   NODE_40_length_14359_cov_7.162903    98.983   \n",
       "92   Gabriel_1153_R   NODE_74_length_3963_cov_27.927277    94.227   \n",
       "93     Gabriel_F_23   NODE_40_length_14359_cov_7.162903    96.325   \n",
       "94     Gabriel_R_77  NODE_40_length_39401_cov_23.095466    95.490   \n",
       "95     Gabriel_R_74  NODE_40_length_39401_cov_23.095466    97.360   \n",
       "96   Gabriel_1154_F   NODE_74_length_3963_cov_27.927277    98.355   \n",
       "97     Gabriel_F_42  NODE_40_length_39401_cov_23.095466    95.150   \n",
       "98     Gabriel_F_37  NODE_40_length_39401_cov_23.095466    94.523   \n",
       "99     Gabriel_F_73  NODE_40_length_39401_cov_23.095466    96.967   \n",
       "100  Gabriel_1189_R   NODE_6_length_238953_cov_6.622376    96.098   \n",
       "101  Gabriel_1132_F   NODE_74_length_3963_cov_27.927277    91.128   \n",
       "102    Gabriel_F_49  NODE_40_length_39401_cov_23.095466    97.436   \n",
       "103    Gabriel_R_22   NODE_6_length_238953_cov_6.622376    95.352   \n",
       "104  Gabriel_1190_R    NODE_51_length_2906_cov_7.795155    97.693   \n",
       "105    Gabriel_R_60  NODE_40_length_39401_cov_23.095466    96.456   \n",
       "106    Gabriel_F_28  NODE_40_length_39401_cov_23.095466    97.428   \n",
       "107    Gabriel_F_56  NODE_40_length_39401_cov_23.095466    91.939   \n",
       "\n",
       "     alignment length  mismatches   gap opens  q. start  q. end  s. start  \\\n",
       "0                1116          20          34        39    1151         1   \n",
       "1                 730           4          10        51     776     17064   \n",
       "2                 406           4           3        33     438     19645   \n",
       "3                 890           7          15        33     909      3556   \n",
       "4                1064          10          21        32    1074         1   \n",
       "5                1044           6          34         9    1051     23247   \n",
       "6                 368           5           0        63     430       368   \n",
       "7                 954          30          26        31     984    138839   \n",
       "8                1052          11          21        32    1080      2387   \n",
       "9                1104           8          26        27    1130      1842   \n",
       "10                853           1          25        35     887     21409   \n",
       "11               1041           4          29        50    1090     18667   \n",
       "12                746           2          21        52     797     23757   \n",
       "13                778           3          23        54     831     23757   \n",
       "14                863           7          37        25     848      1083   \n",
       "15                996           9          28        26    1020     19357   \n",
       "16               1125          26          24        32    1154    118155   \n",
       "17               1146           6          30        29    1174    120091   \n",
       "18                791           3          17        29     818     22450   \n",
       "19                449           2           2        36     482      3251   \n",
       "20                691           2          13        29     719     21574   \n",
       "21               1138          10          15        32    1163     61716   \n",
       "22                974           1          36        15     988     19512   \n",
       "23               1171           5          27        48    1210    118155   \n",
       "24               1058           7          48         4    1060     23137   \n",
       "25                368           5           0        40     407       368   \n",
       "26                780           1          32        29     808      1842   \n",
       "27                647           0          11        37     683     22699   \n",
       "28               1148           3          21        34    1178      3556   \n",
       "29               1088          24          22        51    1135         1   \n",
       "..                ...         ...         ...       ...     ...       ...   \n",
       "78                583           4          19       170     731         1   \n",
       "79                926           1          26        12     936     21484   \n",
       "80               1106           5          20        28    1133      1842   \n",
       "81               1019          10          45        34    1050     19668   \n",
       "82                836           1           2        48     883      3251   \n",
       "83               1021           7          18        22    1042     19272   \n",
       "84                873           0          29        33     905     18976   \n",
       "85                932           2          39        36     967     23757   \n",
       "86                912           2          31        15     926     19668   \n",
       "87               1144           9          16        27    1165    118155   \n",
       "88                890           0          43        37     926     21302   \n",
       "89                446          13          49        52     448     21299   \n",
       "90                916           6          31        34     949     23236   \n",
       "91               1082           1          10        50    1121      1083   \n",
       "92               1126          23          40         9    1133      2356   \n",
       "93               1034           9          27         7    1040      1084   \n",
       "94                776           4          30        29     804     21568   \n",
       "95                606           4          12        34     639     19668   \n",
       "96               1094           6          12        42    1134         1   \n",
       "97                969           6          37        19     987     20028   \n",
       "98                986           6          47         5     986     22344   \n",
       "99                989           3          27        14    1001     17067   \n",
       "100              1179          12          32        33    1211    120091   \n",
       "101               541          27          21        71     591         1   \n",
       "102              1014           6          20        18    1029     21577   \n",
       "103               710          14          19        40     749    120047   \n",
       "104              1127           6          20        26    1151      1842   \n",
       "105               790           1          26        34     823     22061   \n",
       "106              1011           3          21        57    1064     17064   \n",
       "107               980          30          47        41    1018     21880   \n",
       "\n",
       "     s. end         evalue  bit score  \\\n",
       "0      1084   0.000000e+00       1724   \n",
       "1     17787   0.000000e+00       1262   \n",
       "2     20047   0.000000e+00        708   \n",
       "3      2669   0.000000e+00       1507   \n",
       "4      1063   0.000000e+00       1768   \n",
       "5     22238   0.000000e+00       1668   \n",
       "6         1   0.000000e+00        652   \n",
       "7    139766   0.000000e+00       1428   \n",
       "8      1354   0.000000e+00       1746   \n",
       "9       765   0.000000e+00       1827   \n",
       "10    20582   0.000000e+00       1408   \n",
       "11    19677   0.000000e+00       1707   \n",
       "12    23033   0.000000e+00       1230   \n",
       "13    23004   0.000000e+00       1266   \n",
       "14      221   0.000000e+00       1303   \n",
       "15    20325   0.000000e+00       1609   \n",
       "16   119257   0.000000e+00       1779   \n",
       "17   118979   0.000000e+00       1869   \n",
       "18    21676   0.000000e+00       1334   \n",
       "19     2803   0.000000e+00        806   \n",
       "20    20899   0.000000e+00       1168   \n",
       "21    60588   0.000000e+00       1949   \n",
       "22    20448   0.000000e+00       1554   \n",
       "23   119306   0.000000e+00       1960   \n",
       "24    22128   0.000000e+00       1598   \n",
       "25        1   0.000000e+00        652   \n",
       "26     1097   0.000000e+00       1216   \n",
       "27    22064   0.000000e+00       1123   \n",
       "28     2428   0.000000e+00       1962   \n",
       "29     1069   0.000000e+00       1735   \n",
       "..      ...            ...        ...   \n",
       "78      583   0.000000e+00        918   \n",
       "79    20585   0.000000e+00       1530   \n",
       "80      758   0.000000e+00       1879   \n",
       "81    20642   0.000000e+00       1530   \n",
       "82     2418   0.000000e+00       1526   \n",
       "83    20274   0.000000e+00       1731   \n",
       "84    19819   0.000000e+00       1424   \n",
       "85    22869   0.000000e+00       1432   \n",
       "86    20546   0.000000e+00       1459   \n",
       "87   119287   0.000000e+00       1960   \n",
       "88    20456   0.000000e+00       1365   \n",
       "89    20854  2.620000e-122        435   \n",
       "90    22352   0.000000e+00       1458   \n",
       "91        2   0.000000e+00       1929   \n",
       "92     1272   0.000000e+00       1681   \n",
       "93       80   0.000000e+00       1672   \n",
       "94    20824   0.000000e+00       1210   \n",
       "95    20261   0.000000e+00       1020   \n",
       "96     1083   0.000000e+00       1910   \n",
       "97    20955   0.000000e+00       1491   \n",
       "98    21403   0.000000e+00       1478   \n",
       "99    18029   0.000000e+00       1635   \n",
       "100  118947   0.000000e+00       1892   \n",
       "101     540   0.000000e+00        713   \n",
       "102   20582   0.000000e+00       1711   \n",
       "103  119357   0.000000e+00       1110   \n",
       "104     735   0.000000e+00       1919   \n",
       "105   21299   0.000000e+00       1279   \n",
       "106   18054   0.000000e+00       1701   \n",
       "107   20948   0.000000e+00       1327   \n",
       "\n",
       "                                               arquivo  \n",
       "0    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "1    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "2    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "3    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "4    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "5    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "6    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "7    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "8    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "9    /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "10   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "11   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "12   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "13   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "14   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "15   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "16   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "17   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "18   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "19   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "20   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "21   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "22   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "23   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "24   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "25   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "26   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "27   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "28   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "29   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "..                                                 ...  \n",
       "78   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "79   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "80   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "81   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "82   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "83   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "84   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "85   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "86   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "87   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "88   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "89   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "90   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "91   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "92   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "93   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "94   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "95   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "96   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "97   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "98   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "99   /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "100  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "101  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "102  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "103  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "104  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "105  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "106  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "107  /home/tiagoboreli/Documentos/trimopython/blast...  \n",
       "\n",
       "[108 rows x 13 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_f = genomes.set_index('query acc.ver').filter(like = '_F', axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_f.sort_index(inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_f.reset_index(inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_r = genomes.set_index('query acc.ver').filter(like = '_R', axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_r.sort_index(inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_r.reset_index(inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_r.to_csv('genomes_r.csv',sep = ',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes_f.to_csv('genomes_f.csv',sep = ',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "strac = pd.DataFrame({'Primer_f':genomes_f['query acc.ver'],\n",
    "                      'Primer_r':genomes_r['query acc.ver'],\n",
    "                      'Nodes':genomes_f['subject acc.ver'] + '/' + genomes_r['subject acc.ver'],\n",
    "                     'Genoma':genomes_r['arquivo'].apply(lambda x: x.split('/')[-1].split('.')[2].split('_')[1])})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "l = 0\n",
    "i = 0\n",
    "start = []\n",
    "End = []\n",
    "while l < len(genomes_r):\n",
    "    p = [genomes_f['s. start'][i], genomes_f['s. end'][i],genomes_r['s. start'][i], genomes_r['s. end'][i]]\n",
    "    start.append(min(p))\n",
    "    End.append(max(p))\n",
    "    p = []\n",
    "    l += 1\n",
    "    i += 1\n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "strac['Start'] = start\n",
    "strac['End'] = End\n",
    "strac['Length'] = strac['End'] - strac['Start']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "validation = []\n",
    "for items in strac.itertuples():\n",
    "    if items.Nodes.split('/')[0] == items.Nodes.split('/')[1]:\n",
    "        validation.append('True')\n",
    "    else:\n",
    "        validation.append('False')\n",
    "strac['Validation'] = validation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     46\n",
       "False     8\n",
       "Name: Validation, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strac['Validation'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Primer_f</th>\n",
       "      <th>Primer_r</th>\n",
       "      <th>Nodes</th>\n",
       "      <th>Genoma</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Validation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_1122_F</td>\n",
       "      <td>Gabriel_1122_R</td>\n",
       "      <td>NODE_102_length_736_cov_307.748823/NODE_74_len...</td>\n",
       "      <td>genome36</td>\n",
       "      <td>1</td>\n",
       "      <td>3556</td>\n",
       "      <td>3555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_1149_F</td>\n",
       "      <td>Gabriel_1149_R</td>\n",
       "      <td>NODE_102_length_736_cov_307.748823/NODE_74_len...</td>\n",
       "      <td>genome36</td>\n",
       "      <td>1</td>\n",
       "      <td>3556</td>\n",
       "      <td>3555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_1183_F</td>\n",
       "      <td>Gabriel_1183_R</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903/NODE_51_leng...</td>\n",
       "      <td>genome40</td>\n",
       "      <td>221</td>\n",
       "      <td>1842</td>\n",
       "      <td>1621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_1184_F</td>\n",
       "      <td>Gabriel_1184_R</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903/NODE_51_leng...</td>\n",
       "      <td>genome40</td>\n",
       "      <td>2</td>\n",
       "      <td>1842</td>\n",
       "      <td>1840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_1190_F</td>\n",
       "      <td>Gabriel_1190_R</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903/NODE_51_leng...</td>\n",
       "      <td>genome40</td>\n",
       "      <td>87</td>\n",
       "      <td>1842</td>\n",
       "      <td>1755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_F_23</td>\n",
       "      <td>Gabriel_R_23</td>\n",
       "      <td>NODE_40_length_14359_cov_7.162903/NODE_51_leng...</td>\n",
       "      <td>genome40</td>\n",
       "      <td>80</td>\n",
       "      <td>1842</td>\n",
       "      <td>1762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_F_27</td>\n",
       "      <td>Gabriel_R_27</td>\n",
       "      <td>NODE_1_length_357564_cov_9.635492/NODE_40_leng...</td>\n",
       "      <td>genome3</td>\n",
       "      <td>21005</td>\n",
       "      <td>82674</td>\n",
       "      <td>61669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>Gabriel_F_72</td>\n",
       "      <td>Gabriel_R_72</td>\n",
       "      <td>NODE_1_length_357564_cov_9.635492/NODE_40_leng...</td>\n",
       "      <td>genome3</td>\n",
       "      <td>20383</td>\n",
       "      <td>82660</td>\n",
       "      <td>62277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Primer_f        Primer_r  \\\n",
       "Validation                                   \n",
       "False       Gabriel_1122_F  Gabriel_1122_R   \n",
       "False       Gabriel_1149_F  Gabriel_1149_R   \n",
       "False       Gabriel_1183_F  Gabriel_1183_R   \n",
       "False       Gabriel_1184_F  Gabriel_1184_R   \n",
       "False       Gabriel_1190_F  Gabriel_1190_R   \n",
       "False         Gabriel_F_23    Gabriel_R_23   \n",
       "False         Gabriel_F_27    Gabriel_R_27   \n",
       "False         Gabriel_F_72    Gabriel_R_72   \n",
       "\n",
       "                                                        Nodes    Genoma  \\\n",
       "Validation                                                                \n",
       "False       NODE_102_length_736_cov_307.748823/NODE_74_len...  genome36   \n",
       "False       NODE_102_length_736_cov_307.748823/NODE_74_len...  genome36   \n",
       "False       NODE_40_length_14359_cov_7.162903/NODE_51_leng...  genome40   \n",
       "False       NODE_40_length_14359_cov_7.162903/NODE_51_leng...  genome40   \n",
       "False       NODE_40_length_14359_cov_7.162903/NODE_51_leng...  genome40   \n",
       "False       NODE_40_length_14359_cov_7.162903/NODE_51_leng...  genome40   \n",
       "False       NODE_1_length_357564_cov_9.635492/NODE_40_leng...   genome3   \n",
       "False       NODE_1_length_357564_cov_9.635492/NODE_40_leng...   genome3   \n",
       "\n",
       "            Start    End  Length  \n",
       "Validation                        \n",
       "False           1   3556    3555  \n",
       "False           1   3556    3555  \n",
       "False         221   1842    1621  \n",
       "False           2   1842    1840  \n",
       "False          87   1842    1755  \n",
       "False          80   1842    1762  \n",
       "False       21005  82674   61669  \n",
       "False       20383  82660   62277  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strac.set_index('Validation').loc['False']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "strac.to_csv('csv_to_strac.csv', sep = ',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.system('mkdir seqs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "for items in strac.itertuples():\n",
    "    file = open(f'{items.Genoma}_seq.fasta','a')\n",
    "    \n",
    "    for fasta in SeqIO.parse(f'{items.Genoma}.fasta','fasta'):\n",
    "        if fasta.id == items.Nodes.split('/')[0]:\n",
    "            sequence = str(fasta.seq)\n",
    "            start = int(items.Start)\n",
    "            end = int(items.End)\n",
    "            sequence_final = (sequence[start-1:end])\n",
    "            \n",
    "            \n",
    "            \n",
    "            \n",
    "            \n",
    "\n",
    "            file.write(f'>{items.Genoma}_clone_{items.Primer_f[8:12]}\\n')\n",
    "            file.write(f\"{sequence_final} \\n\")\n",
    "    file.close()\n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mv /home/tiagoboreli/Documentos/trimopython/*seq.fasta /home/tiagoboreli/Documentos/trimopython/seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "caminho = '/home/tiagoboreli/Documentos/trimopython/seqs'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "customfunc.create(status = 'yes', path = caminho)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "abricate iniciado\n",
      "abricate finalizado\n",
      "Foram analisados 3 genomas\n"
     ]
    }
   ],
   "source": [
    "customfunc.execute(status = 'yes', path = caminho, ext = 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:29:50] This is prokka 1.11\n",
      "[15:29:50] Written by Torsten Seemann <torsten.seemann@monash.edu>\n",
      "[15:29:50] Victorian Bioinformatics Consortium - http://www.vicbioinformatics.com\n",
      "[15:29:50] Local time is Fri Feb 21 15:29:50 2020\n",
      "[15:29:50] You are tiagoboreli\n",
      "[15:29:50] Operating system is linux\n",
      "[15:29:50] You have BioPerl 1.007002\n",
      "[15:29:50] System has 8 cores.\n",
      "[15:29:50] Will use maximum of 8 cores.\n",
      "[15:29:50] Annotating as >>> Bacteria <<<\n",
      "[15:29:50] Creating new output folder: prokka36\n",
      "[15:29:50] Running: mkdir -p prokka36\n",
      "[15:29:50] Using filename prefix: PROKKA_02212020.XXX\n",
      "[15:29:50] Setting HMMER_NCPU=1\n",
      "[15:29:50] Writing log to: prokka36/PROKKA_02212020.log\n",
      "[15:29:50] Command: /home/tiagoboreli/Documentos/trimopython/seqs/genome36_seq.fasta --outdir prokka36\n",
      "[15:29:50] Looking for 'aragorn' - found /home/tiagoboreli/anaconda3/bin/aragorn\n",
      "[15:29:50] Determined aragorn version is 1.2\n",
      "[15:29:50] Looking for 'barrnap' - found /home/tiagoboreli/anaconda3/bin/barrnap\n",
      "[15:29:50] Determined barrnap version is 0.9\n",
      "[15:29:50] Looking for 'blastp' - found /home/tiagoboreli/anaconda3/bin/blastp\n",
      "[15:29:50] Determined blastp version is 2.9\n",
      "[15:29:50] Looking for 'cmpress' - found /home/tiagoboreli/anaconda3/bin/cmpress\n",
      "[15:29:50] Determined cmpress version is 1.1\n",
      "[15:29:50] Looking for 'cmscan' - found /home/tiagoboreli/anaconda3/bin/cmscan\n",
      "[15:29:50] Determined cmscan version is 1.1\n",
      "[15:29:50] Looking for 'egrep' - found /bin/egrep\n",
      "[15:29:50] Looking for 'find' - found /usr/bin/find\n",
      "[15:29:50] Looking for 'grep' - found /bin/grep\n",
      "[15:29:50] Looking for 'hmmpress' - found /home/tiagoboreli/anaconda3/bin/hmmpress\n",
      "[15:29:50] Determined hmmpress version is 3.2\n",
      "[15:29:50] Looking for 'hmmscan' - found /home/tiagoboreli/anaconda3/bin/hmmscan\n",
      "[15:29:50] Determined hmmscan version is 3.2\n",
      "[15:29:50] Looking for 'less' - found /usr/bin/less\n",
      "[15:29:50] Looking for 'makeblastdb' - found /home/tiagoboreli/anaconda3/bin/makeblastdb\n",
      "[15:29:50] Determined makeblastdb version is 2.9\n",
      "[15:29:50] Looking for 'minced' - found /home/tiagoboreli/anaconda3/bin/minced\n",
      "[15:29:50] Determined minced version is 4.2\n",
      "[15:29:50] Looking for 'parallel' - found /home/tiagoboreli/anaconda3/bin/parallel\n",
      "[15:29:50] Determined parallel version is 20160622\n",
      "[15:29:50] Looking for 'prodigal' - found /home/tiagoboreli/anaconda3/bin/prodigal\n",
      "[15:29:50] Determined prodigal version is 2.6\n",
      "[15:29:50] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiagoboreli/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[15:29:50] Looking for 'sed' - found /bin/sed\n",
      "[15:29:50] Looking for 'tbl2asn' - found /home/tiagoboreli/anaconda3/bin/tbl2asn\n",
      "[15:29:50] Determined tbl2asn version is 25.8\n",
      "[15:29:50] Using genetic code table 11.\n",
      "[15:29:50] Loading and checking input file: /home/tiagoboreli/Documentos/trimopython/seqs/genome36_seq.fasta\n",
      "[15:29:50] Wrote 13 contigs\n",
      "[15:29:50] Predicting tRNAs and tmRNAs\n",
      "[15:29:50] Running: aragorn -gc11  -w prokka36\\/PROKKA_02212020\\.fna\n",
      "[15:29:50] Found 0 tRNAs\n",
      "[15:29:50] Predicting Ribosomal RNAs\n",
      "[15:29:50] Running Barrnap with 8 threads\n",
      "[15:29:50] Found 0 rRNAs\n",
      "[15:29:50] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[15:29:50] Total of 0 tRNA + rRNA features\n",
      "[15:29:50] Searching for CRISPR repeats\n",
      "[15:29:50] Found 0 CRISPRs\n",
      "[15:29:50] Predicting coding sequences\n",
      "[15:29:50] Contigs total 28339 bp, so using meta mode\n",
      "[15:29:50] Running: prodigal -i prokka36\\/PROKKA_02212020\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[15:29:51] Found 23 CDS\n",
      "[15:29:51] Connecting features back to sequences\n",
      "[15:29:51] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[15:29:51] Annotating CDS, please be patient.\n",
      "[15:29:51] Will use 8 CPUs for similarity searching.\n",
      "[15:29:51] There are still 23 unannotated CDS left (started with 23)\n",
      "[15:29:51] Will use blast to search against /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[15:29:51] Running: cat prokka36\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 316 --recstart '>' --pipe blastp -query - -db /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > prokka36\\/sprot\\.blast 2> /dev/null\n",
      "[15:29:51] Deleting unwanted file: prokka36/sprot.faa\n",
      "[15:29:51] Deleting unwanted file: prokka36/sprot.blast\n",
      "[15:29:51] There are still 10 unannotated CDS left (started with 23)\n",
      "[15:29:51] Will use hmmer3 to search against /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[15:29:51] Running: cat prokka36\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 71 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > prokka36\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[15:29:52] Deleting unwanted file: prokka36/HAMAP.hmm.faa\n",
      "[15:29:52] Deleting unwanted file: prokka36/HAMAP.hmm.hmmer3\n",
      "[15:29:52] Labelling remaining 10 proteins as 'hypothetical protein'\n",
      "[15:29:52] Found 2 unique /gene codes.\n",
      "[15:29:52] Fixed 2 duplicate /gene - ftsI_1 ftsI_2\n",
      "[15:29:52] Fixed 11 duplicate /gene - bla_1 bla_2 bla_3 bla_4 bla_5 bla_6 bla_7 bla_8 bla_9 bla_10 bla_11\n",
      "[15:29:52] Fixed 2 colliding /gene names.\n",
      "[15:29:52] Adding /locus_tag identifiers\n",
      "[15:29:52] Assigned 23 locus_tags to CDS and RNA features.\n",
      "[15:29:52] Writing outputs to prokka36/\n",
      "[15:29:52] Generating annotation statistics file\n",
      "[15:29:52] Generating Genbank and Sequin files\n",
      "[15:29:52] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.11 from http://www.vicbioinformatics.com' -Z prokka36\\/PROKKA_02212020\\.err -i prokka36\\/PROKKA_02212020\\.fsa 2> /dev/null\n",
      "[15:29:52] Deleting unwanted file: prokka36/errorsummary.val\n",
      "[15:29:52] Deleting unwanted file: prokka36/PROKKA_02212020.dr\n",
      "[15:29:52] Deleting unwanted file: prokka36/PROKKA_02212020.fixedproducts\n",
      "[15:29:52] Deleting unwanted file: prokka36/PROKKA_02212020.ecn\n",
      "[15:29:52] Deleting unwanted file: prokka36/PROKKA_02212020.val\n",
      "[15:29:52] Repairing broken .GBK output that tbl2asn produces...\n",
      "[15:29:52] Running: sed 's/COORDINATES: profile/COORDINATES:profile/' < prokka36\\/PROKKA_02212020\\.gbf > prokka36\\/PROKKA_02212020\\.gbk\n",
      "[15:29:52] Deleting unwanted file: prokka36/PROKKA_02212020.gbf\n",
      "[15:29:52] Output files:\n",
      "[15:29:52] prokka36/PROKKA_02212020.sqn\n",
      "[15:29:52] prokka36/PROKKA_02212020.fna\n",
      "[15:29:52] prokka36/PROKKA_02212020.fsa\n",
      "[15:29:52] prokka36/PROKKA_02212020.txt\n",
      "[15:29:52] prokka36/PROKKA_02212020.log\n",
      "[15:29:52] prokka36/PROKKA_02212020.faa\n",
      "[15:29:52] prokka36/PROKKA_02212020.ffn\n",
      "[15:29:52] prokka36/PROKKA_02212020.gff\n",
      "[15:29:52] prokka36/PROKKA_02212020.gbk\n",
      "[15:29:52] prokka36/PROKKA_02212020.tbl\n",
      "[15:29:52] prokka36/PROKKA_02212020.err\n",
      "[15:29:52] Walltime used: 0.03 minutes\n",
      "[15:29:52] If you use this result please cite the Prokka paper:\n",
      "[15:29:52] Seemann T (2014) Prokka: rapid prokaryotic genome annotation. Bioinformatics. 30(14):2068-9.\n",
      "[15:29:52] Type 'prokka --citation' for more details.\n",
      "[15:29:52] Thank you, come again.\n"
     ]
    }
   ],
   "source": [
    "!prokka /home/tiagoboreli/Documentos/trimopython/seqs/genome36_seq.fasta --outdir prokka36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:29:57] This is prokka 1.11\n",
      "[15:29:57] Written by Torsten Seemann <torsten.seemann@monash.edu>\n",
      "[15:29:57] Victorian Bioinformatics Consortium - http://www.vicbioinformatics.com\n",
      "[15:29:57] Local time is Fri Feb 21 15:29:57 2020\n",
      "[15:29:57] You are tiagoboreli\n",
      "[15:29:57] Operating system is linux\n",
      "[15:29:57] You have BioPerl 1.007002\n",
      "[15:29:57] System has 8 cores.\n",
      "[15:29:57] Will use maximum of 8 cores.\n",
      "[15:29:57] Annotating as >>> Bacteria <<<\n",
      "[15:29:57] Creating new output folder: prokka40\n",
      "[15:29:57] Running: mkdir -p prokka40\n",
      "[15:29:57] Using filename prefix: PROKKA_02212020.XXX\n",
      "[15:29:57] Setting HMMER_NCPU=1\n",
      "[15:29:57] Writing log to: prokka40/PROKKA_02212020.log\n",
      "[15:29:57] Command: /home/tiagoboreli/Documentos/trimopython/seqs/genome40_seq.fasta --outdir prokka40\n",
      "[15:29:57] Looking for 'aragorn' - found /home/tiagoboreli/anaconda3/bin/aragorn\n",
      "[15:29:57] Determined aragorn version is 1.2\n",
      "[15:29:57] Looking for 'barrnap' - found /home/tiagoboreli/anaconda3/bin/barrnap\n",
      "[15:29:57] Determined barrnap version is 0.9\n",
      "[15:29:57] Looking for 'blastp' - found /home/tiagoboreli/anaconda3/bin/blastp\n",
      "[15:29:57] Determined blastp version is 2.9\n",
      "[15:29:57] Looking for 'cmpress' - found /home/tiagoboreli/anaconda3/bin/cmpress\n",
      "[15:29:57] Determined cmpress version is 1.1\n",
      "[15:29:57] Looking for 'cmscan' - found /home/tiagoboreli/anaconda3/bin/cmscan\n",
      "[15:29:57] Determined cmscan version is 1.1\n",
      "[15:29:57] Looking for 'egrep' - found /bin/egrep\n",
      "[15:29:57] Looking for 'find' - found /usr/bin/find\n",
      "[15:29:57] Looking for 'grep' - found /bin/grep\n",
      "[15:29:57] Looking for 'hmmpress' - found /home/tiagoboreli/anaconda3/bin/hmmpress\n",
      "[15:29:57] Determined hmmpress version is 3.2\n",
      "[15:29:57] Looking for 'hmmscan' - found /home/tiagoboreli/anaconda3/bin/hmmscan\n",
      "[15:29:57] Determined hmmscan version is 3.2\n",
      "[15:29:57] Looking for 'less' - found /usr/bin/less\n",
      "[15:29:57] Looking for 'makeblastdb' - found /home/tiagoboreli/anaconda3/bin/makeblastdb\n",
      "[15:29:57] Determined makeblastdb version is 2.9\n",
      "[15:29:57] Looking for 'minced' - found /home/tiagoboreli/anaconda3/bin/minced\n",
      "[15:29:57] Determined minced version is 4.2\n",
      "[15:29:57] Looking for 'parallel' - found /home/tiagoboreli/anaconda3/bin/parallel\n",
      "[15:29:57] Determined parallel version is 20160622\n",
      "[15:29:57] Looking for 'prodigal' - found /home/tiagoboreli/anaconda3/bin/prodigal\n",
      "[15:29:57] Determined prodigal version is 2.6\n",
      "[15:29:57] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiagoboreli/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[15:29:57] Looking for 'sed' - found /bin/sed\n",
      "[15:29:57] Looking for 'tbl2asn' - found /home/tiagoboreli/anaconda3/bin/tbl2asn\n",
      "[15:29:57] Determined tbl2asn version is 25.8\n",
      "[15:29:57] Using genetic code table 11.\n",
      "[15:29:57] Loading and checking input file: /home/tiagoboreli/Documentos/trimopython/seqs/genome40_seq.fasta\n",
      "[15:29:57] Wrote 8 contigs\n",
      "[15:29:57] Predicting tRNAs and tmRNAs\n",
      "[15:29:57] Running: aragorn -gc11  -w prokka40\\/PROKKA_02212020\\.fna\n",
      "[15:29:57] Found 0 tRNAs\n",
      "[15:29:57] Predicting Ribosomal RNAs\n",
      "[15:29:57] Running Barrnap with 8 threads\n",
      "[15:29:57] Found 0 rRNAs\n",
      "[15:29:57] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[15:29:57] Total of 0 tRNA + rRNA features\n",
      "[15:29:57] Searching for CRISPR repeats\n",
      "[15:29:58] Found 0 CRISPRs\n",
      "[15:29:58] Predicting coding sequences\n",
      "[15:29:58] Contigs total 16291 bp, so using meta mode\n",
      "[15:29:58] Running: prodigal -i prokka40\\/PROKKA_02212020\\.fna -c -m -g 11 -p meta -f sco -q\n",
      "[15:29:58] Found 20 CDS\n",
      "[15:29:58] Connecting features back to sequences\n",
      "[15:29:58] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[15:29:58] Annotating CDS, please be patient.\n",
      "[15:29:58] Will use 8 CPUs for similarity searching.\n",
      "[15:29:58] There are still 20 unannotated CDS left (started with 20)\n",
      "[15:29:58] Will use blast to search against /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[15:29:58] Running: cat prokka40\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 244 --recstart '>' --pipe blastp -query - -db /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > prokka40\\/sprot\\.blast 2> /dev/null\n",
      "[15:29:58] Modify product: Probable chromate transport protein => putative chromate transport protein\n",
      "[15:29:58] Modify product: Probable chromate transport protein => putative chromate transport protein\n",
      "[15:29:58] Cleaned 2 /product names\n",
      "[15:29:58] Deleting unwanted file: prokka40/sprot.faa\n",
      "[15:29:58] Deleting unwanted file: prokka40/sprot.blast\n",
      "[15:29:58] There are still 7 unannotated CDS left (started with 20)\n",
      "[15:29:58] Will use hmmer3 to search against /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[15:29:58] Running: cat prokka40\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 44 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > prokka40\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[15:29:59] Deleting unwanted file: prokka40/HAMAP.hmm.faa\n",
      "[15:29:59] Deleting unwanted file: prokka40/HAMAP.hmm.hmmer3\n",
      "[15:29:59] Labelling remaining 6 proteins as 'hypothetical protein'\n",
      "[15:29:59] Found 7 unique /gene codes.\n",
      "[15:29:59] Fixed 4 duplicate /gene - ampC_1 ampC_2 ampC_3 ampC_4\n",
      "[15:29:59] Fixed 2 duplicate /gene - srpC_1 srpC_2\n",
      "[15:29:59] Fixed 4 duplicate /gene - folP_1 folP_2 folP_3 folP_4\n",
      "[15:29:59] Fixed 3 colliding /gene names.\n",
      "[15:29:59] Adding /locus_tag identifiers\n",
      "[15:29:59] Assigned 20 locus_tags to CDS and RNA features.\n",
      "[15:29:59] Writing outputs to prokka40/\n",
      "[15:29:59] Generating annotation statistics file\n",
      "[15:29:59] Generating Genbank and Sequin files\n",
      "[15:29:59] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.11 from http://www.vicbioinformatics.com' -Z prokka40\\/PROKKA_02212020\\.err -i prokka40\\/PROKKA_02212020\\.fsa 2> /dev/null\n",
      "[15:29:59] Deleting unwanted file: prokka40/errorsummary.val\n",
      "[15:29:59] Deleting unwanted file: prokka40/PROKKA_02212020.dr\n",
      "[15:29:59] Deleting unwanted file: prokka40/PROKKA_02212020.fixedproducts\n",
      "[15:29:59] Deleting unwanted file: prokka40/PROKKA_02212020.ecn\n",
      "[15:29:59] Deleting unwanted file: prokka40/PROKKA_02212020.val\n",
      "[15:29:59] Repairing broken .GBK output that tbl2asn produces...\n",
      "[15:29:59] Running: sed 's/COORDINATES: profile/COORDINATES:profile/' < prokka40\\/PROKKA_02212020\\.gbf > prokka40\\/PROKKA_02212020\\.gbk\n",
      "[15:29:59] Deleting unwanted file: prokka40/PROKKA_02212020.gbf\n",
      "[15:29:59] Output files:\n",
      "[15:29:59] prokka40/PROKKA_02212020.sqn\n",
      "[15:29:59] prokka40/PROKKA_02212020.fna\n",
      "[15:29:59] prokka40/PROKKA_02212020.fsa\n",
      "[15:29:59] prokka40/PROKKA_02212020.txt\n",
      "[15:29:59] prokka40/PROKKA_02212020.log\n",
      "[15:29:59] prokka40/PROKKA_02212020.faa\n",
      "[15:29:59] prokka40/PROKKA_02212020.ffn\n",
      "[15:29:59] prokka40/PROKKA_02212020.gff\n",
      "[15:29:59] prokka40/PROKKA_02212020.gbk\n",
      "[15:29:59] prokka40/PROKKA_02212020.tbl\n",
      "[15:29:59] prokka40/PROKKA_02212020.err\n",
      "[15:29:59] Walltime used: 0.03 minutes\n",
      "[15:29:59] If you use this result please cite the Prokka paper:\n",
      "[15:29:59] Seemann T (2014) Prokka: rapid prokaryotic genome annotation. Bioinformatics. 30(14):2068-9.\n",
      "[15:29:59] Type 'prokka --citation' for more details.\n",
      "[15:29:59] Thank you, come again.\n"
     ]
    }
   ],
   "source": [
    "!prokka /home/tiagoboreli/Documentos/trimopython/seqs/genome40_seq.fasta --outdir prokka40"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[15:30:04] This is prokka 1.11\n",
      "[15:30:04] Written by Torsten Seemann <torsten.seemann@monash.edu>\n",
      "[15:30:04] Victorian Bioinformatics Consortium - http://www.vicbioinformatics.com\n",
      "[15:30:04] Local time is Fri Feb 21 15:30:04 2020\n",
      "[15:30:04] You are tiagoboreli\n",
      "[15:30:04] Operating system is linux\n",
      "[15:30:04] You have BioPerl 1.007002\n",
      "[15:30:04] System has 8 cores.\n",
      "[15:30:04] Will use maximum of 8 cores.\n",
      "[15:30:04] Annotating as >>> Bacteria <<<\n",
      "[15:30:04] Creating new output folder: prokka03\n",
      "[15:30:04] Running: mkdir -p prokka03\n",
      "[15:30:04] Using filename prefix: PROKKA_02212020.XXX\n",
      "[15:30:04] Setting HMMER_NCPU=1\n",
      "[15:30:04] Writing log to: prokka03/PROKKA_02212020.log\n",
      "[15:30:04] Command: /home/tiagoboreli/Documentos/trimopython/seqs/genome3_seq.fasta --outdir prokka03\n",
      "[15:30:04] Looking for 'aragorn' - found /home/tiagoboreli/anaconda3/bin/aragorn\n",
      "[15:30:04] Determined aragorn version is 1.2\n",
      "[15:30:04] Looking for 'barrnap' - found /home/tiagoboreli/anaconda3/bin/barrnap\n",
      "[15:30:04] Determined barrnap version is 0.9\n",
      "[15:30:04] Looking for 'blastp' - found /home/tiagoboreli/anaconda3/bin/blastp\n",
      "[15:30:04] Determined blastp version is 2.9\n",
      "[15:30:04] Looking for 'cmpress' - found /home/tiagoboreli/anaconda3/bin/cmpress\n",
      "[15:30:04] Determined cmpress version is 1.1\n",
      "[15:30:04] Looking for 'cmscan' - found /home/tiagoboreli/anaconda3/bin/cmscan\n",
      "[15:30:04] Determined cmscan version is 1.1\n",
      "[15:30:04] Looking for 'egrep' - found /bin/egrep\n",
      "[15:30:04] Looking for 'find' - found /usr/bin/find\n",
      "[15:30:04] Looking for 'grep' - found /bin/grep\n",
      "[15:30:04] Looking for 'hmmpress' - found /home/tiagoboreli/anaconda3/bin/hmmpress\n",
      "[15:30:04] Determined hmmpress version is 3.2\n",
      "[15:30:04] Looking for 'hmmscan' - found /home/tiagoboreli/anaconda3/bin/hmmscan\n",
      "[15:30:04] Determined hmmscan version is 3.2\n",
      "[15:30:04] Looking for 'less' - found /usr/bin/less\n",
      "[15:30:04] Looking for 'makeblastdb' - found /home/tiagoboreli/anaconda3/bin/makeblastdb\n",
      "[15:30:04] Determined makeblastdb version is 2.9\n",
      "[15:30:04] Looking for 'minced' - found /home/tiagoboreli/anaconda3/bin/minced\n",
      "[15:30:04] Determined minced version is 4.2\n",
      "[15:30:04] Looking for 'parallel' - found /home/tiagoboreli/anaconda3/bin/parallel\n",
      "[15:30:04] Determined parallel version is 20160622\n",
      "[15:30:04] Looking for 'prodigal' - found /home/tiagoboreli/anaconda3/bin/prodigal\n",
      "[15:30:04] Determined prodigal version is 2.6\n",
      "[15:30:04] Looking for 'prokka-genbank_to_fasta_db' - found /home/tiagoboreli/anaconda3/bin/prokka-genbank_to_fasta_db\n",
      "[15:30:04] Looking for 'sed' - found /bin/sed\n",
      "[15:30:04] Looking for 'tbl2asn' - found /home/tiagoboreli/anaconda3/bin/tbl2asn\n",
      "[15:30:04] Determined tbl2asn version is 25.8\n",
      "[15:30:04] Using genetic code table 11.\n",
      "[15:30:04] Loading and checking input file: /home/tiagoboreli/Documentos/trimopython/seqs/genome3_seq.fasta\n",
      "[15:30:04] Wrote 33 contigs\n",
      "[15:30:04] Predicting tRNAs and tmRNAs\n",
      "[15:30:04] Running: aragorn -gc11  -w prokka03\\/PROKKA_02212020\\.fna\n",
      "[15:30:04] Found 0 tRNAs\n",
      "[15:30:04] Predicting Ribosomal RNAs\n",
      "[15:30:04] Running Barrnap with 8 threads\n",
      "[15:30:05] Found 0 rRNAs\n",
      "[15:30:05] Skipping ncRNA search, enable with --rfam if desired.\n",
      "[15:30:05] Total of 0 tRNA + rRNA features\n",
      "[15:30:05] Searching for CRISPR repeats\n",
      "[15:30:05] Found 0 CRISPRs\n",
      "[15:30:05] Predicting coding sequences\n",
      "[15:30:05] Contigs total 225236 bp, so using single mode\n",
      "[15:30:05] Running: prodigal -i prokka03\\/PROKKA_02212020\\.fna -c -m -g 11 -p single -f sco -q\n",
      "[15:30:05] Found 292 CDS\n",
      "[15:30:05] Connecting features back to sequences\n",
      "[15:30:05] Not using genus-specific database. Try --usegenus to enable it.\n",
      "[15:30:05] Annotating CDS, please be patient.\n",
      "[15:30:05] Will use 8 CPUs for similarity searching.\n",
      "[15:30:05] There are still 292 unannotated CDS left (started with 292)\n",
      "[15:30:05] Will use blast to search against /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot with 8 CPUs\n",
      "[15:30:05] Running: cat prokka03\\/sprot\\.faa | parallel --gnu --plain -j 8 --block 4315 --recstart '>' --pipe blastp -query - -db /home/tiagoboreli/anaconda3/bin/../db/kingdom/Bacteria/sprot -evalue 1e-06 -num_threads 1 -num_descriptions 1 -num_alignments 1 -seg no > prokka03\\/sprot\\.blast 2> /dev/null\n",
      "[15:30:08] Modify product: Probable 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD => putative 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD\n",
      "[15:30:08] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE\n",
      "[15:30:08] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF\n",
      "[15:30:08] Modify product: Uncharacterized Nudix hydrolase YfcD => putative Nudix hydrolase YfcD\n",
      "[15:30:08] Modify product: Epimerase family protein SA0724 => Epimerase family protein\n",
      "[15:30:08] Modify product: Uncharacterized Nudix hydrolase YfcD => putative Nudix hydrolase YfcD\n",
      "[15:30:08] Modify product: Epimerase family protein SA0724 => Epimerase family protein\n",
      "[15:30:08] Modify product: Probable 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD => putative 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD\n",
      "[15:30:08] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnE\n",
      "[15:30:08] Modify product: Probable 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF => putative 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol flippase subunit ArnF\n",
      "[15:30:08] Cleaned 10 /product names\n",
      "[15:30:08] Deleting unwanted file: prokka03/sprot.faa\n",
      "[15:30:08] Deleting unwanted file: prokka03/sprot.blast\n",
      "[15:30:08] There are still 153 unannotated CDS left (started with 292)\n",
      "[15:30:08] Will use hmmer3 to search against /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm with 8 CPUs\n",
      "[15:30:08] Running: cat prokka03\\/HAMAP\\.hmm\\.faa | parallel --gnu --plain -j 8 --block 1647 --recstart '>' --pipe hmmscan --noali --notextw --acc -E 1e-06 --cpu 1 /home/tiagoboreli/anaconda3/bin/../db/hmm/HAMAP.hmm /dev/stdin > prokka03\\/HAMAP\\.hmm\\.hmmer3 2> /dev/null\n",
      "[15:30:10] Deleting unwanted file: prokka03/HAMAP.hmm.faa\n",
      "[15:30:10] Deleting unwanted file: prokka03/HAMAP.hmm.hmmer3\n",
      "[15:30:10] Labelling remaining 133 proteins as 'hypothetical protein'\n",
      "[15:30:10] Possible /pseudo 'Protease ElaD' at genome3_clone_F_27 position 13249\n",
      "[15:30:10] Possible /pseudo 'Protease ElaD' at genome3_clone_F_72 position 13871\n",
      "[15:30:10] Found 56 unique /gene codes.\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoI_1 nuoI_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoM_1 nuoM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfcF_1 yfcF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menB_1 menB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hisJ_1 hisJ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - purF_1 purF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menH_1 menH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ackA_1 ackA_2\n",
      "[15:30:10] Fixed 29 duplicate /gene - bla_1 bla_2 bla_3 bla_4 bla_5 bla_6 bla_7 bla_8 bla_9 bla_10 bla_11 bla_12 bla_13 bla_14 bla_15 bla_16 bla_17 bla_18 bla_19 bla_20 bla_21 bla_22 bla_23 bla_24 bla_25 bla_26 bla_27 bla_28 bla_29\n",
      "[15:30:10] Fixed 2 duplicate /gene - cvpA_1 cvpA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoN_1 nuoN_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoC_1 nuoC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoG_1 nuoG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfcD_1 yfcD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - gltC_1 gltC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - argT_1 argT_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hisQ_1 hisQ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menE_1 menE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menF_1 menF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - folC_1 folC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoE_1 nuoE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ndhC_1 ndhC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - accD_1 accD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoB_1 nuoB_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfcG_1 yfcG_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoH_1 nuoH_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - arnE_1 arnE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menC_1 menC_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoL_1 nuoL_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoJ_1 nuoJ_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - sdcS_1 sdcS_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - arnD_1 arnD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - arnF_1 arnF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoK_1 nuoK_2\n",
      "[15:30:10] Fixed 4 duplicate /gene - elaD_1 elaD_2 elaD_3 elaD_4\n",
      "[15:30:10] Fixed 20 duplicate /gene - dnaA_1 dnaA_2 dnaA_3 dnaA_4 dnaA_5 dnaA_6 dnaA_7 dnaA_8 dnaA_9 dnaA_10 dnaA_11 dnaA_12 dnaA_13 dnaA_14 dnaA_15 dnaA_16 dnaA_17 dnaA_18 dnaA_19 dnaA_20\n",
      "[15:30:10] Fixed 2 duplicate /gene - pmrD_1 pmrD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - ubiX_1 ubiX_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfbT_1 yfbT_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - rbn_1 rbn_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - menD_1 menD_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - alaA_1 alaA_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - arnT_1 arnT_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - folX_1 folX_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfcE_1 yfcE_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - yfbR_1 yfbR_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hisP_1 hisP_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - hisM_1 hisM_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - nuoF_1 nuoF_2\n",
      "[15:30:10] Fixed 2 duplicate /gene - pta_1 pta_2\n",
      "[15:30:10] Fixed 50 colliding /gene names.\n",
      "[15:30:10] Adding /locus_tag identifiers\n",
      "[15:30:10] Assigned 292 locus_tags to CDS and RNA features.\n",
      "[15:30:10] Writing outputs to prokka03/\n",
      "[15:30:11] Generating annotation statistics file\n",
      "[15:30:11] Generating Genbank and Sequin files\n",
      "[15:30:11] Running: tbl2asn -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.11 from http://www.vicbioinformatics.com' -Z prokka03\\/PROKKA_02212020\\.err -i prokka03\\/PROKKA_02212020\\.fsa 2> /dev/null\n",
      "[15:30:11] Deleting unwanted file: prokka03/errorsummary.val\n",
      "[15:30:11] Deleting unwanted file: prokka03/PROKKA_02212020.dr\n",
      "[15:30:11] Deleting unwanted file: prokka03/PROKKA_02212020.fixedproducts\n",
      "[15:30:11] Deleting unwanted file: prokka03/PROKKA_02212020.ecn\n",
      "[15:30:11] Deleting unwanted file: prokka03/PROKKA_02212020.val\n",
      "[15:30:11] Repairing broken .GBK output that tbl2asn produces...\n",
      "[15:30:11] Running: sed 's/COORDINATES: profile/COORDINATES:profile/' < prokka03\\/PROKKA_02212020\\.gbf > prokka03\\/PROKKA_02212020\\.gbk\n",
      "[15:30:11] Deleting unwanted file: prokka03/PROKKA_02212020.gbf\n",
      "[15:30:11] Output files:\n",
      "[15:30:11] prokka03/PROKKA_02212020.sqn\n",
      "[15:30:11] prokka03/PROKKA_02212020.fna\n",
      "[15:30:11] prokka03/PROKKA_02212020.fsa\n",
      "[15:30:11] prokka03/PROKKA_02212020.txt\n",
      "[15:30:11] prokka03/PROKKA_02212020.log\n",
      "[15:30:11] prokka03/PROKKA_02212020.faa\n",
      "[15:30:11] prokka03/PROKKA_02212020.ffn\n",
      "[15:30:11] prokka03/PROKKA_02212020.gff\n",
      "[15:30:11] prokka03/PROKKA_02212020.gbk\n",
      "[15:30:11] prokka03/PROKKA_02212020.tbl\n",
      "[15:30:11] prokka03/PROKKA_02212020.err\n",
      "[15:30:11] Walltime used: 0.12 minutes\n",
      "[15:30:11] If you use this result please cite the Prokka paper:\n",
      "[15:30:11] Seemann T (2014) Prokka: rapid prokaryotic genome annotation. Bioinformatics. 30(14):2068-9.\n",
      "[15:30:11] Type 'prokka --citation' for more details.\n",
      "[15:30:11] Share and enjoy!\n"
     ]
    }
   ],
   "source": [
    "!prokka /home/tiagoboreli/Documentos/trimopython/seqs/genome3_seq.fasta --outdir prokka03"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
